pennylane/spin/spin_hamiltonian.py

```diff
@@ -94,3 +97,184 @@ def transverse_ising(
+
+    return hamiltonian
+
+
+def fermihubbard(
+    lattice,
+    n_cells,
+    hopping=None,
+    interaction=1.0,
+    boundary_condition=False,
+    neighbour_order=1,
+    mapping="jordan_wigner",
+):
+    r"""Generates the Hubbard model on a lattice.
+
+    The Hamiltonian is represented as:
+
+    .. math::
+
+        \hat{H} = -t\sum_{<i,j>, \sigma}(c_{i\sigma}^{\dagger}c_{j\sigma}) + U\sum_{i}n_{i \uparrow} n_{i\downarrow}
+
+    where ``t`` is the hopping term representing the kinetic energy of electrons, and ``U`` is the on-site Coulomb interaction,
+    representing the repulsion between electrons.
+
+    Args:
+       lattice (str): Shape of the lattice. Input Values can be ``'chain'``, ``'square'``,
+                      ``'rectangle'``, ``'honeycomb'``, ``'triangle'``, or ``'kagome'``.
+       n_cells (List[int]): Number of cells in each direction of the grid.
+       hopping (float or List[float] or List[math.array(float)]):Hopping interaction between spins, it can be a
+                      list of length equal to ``neighbour_order`` or a square matrix of size
+                      ``(num_spins, num_spins)``. Default value is [1.0].
+       interaction (float or List[float]): Coulomb interaction between spins, it can be a constant or a list of length ``num_spins``.
+       boundary_condition (bool or list[bool]): Defines boundary conditions different lattice axes,
+           default is ``False`` indicating open boundary condition.
+       neighbour_order (int): Specifies the interaction level for neighbors within the lattice.
+                       Default is 1 (nearest neighbour).
+       mapping (str): Specifies the fermion-to-qubit mapping. Input values can be
+                      ``'jordan_wigner'``, ``'parity'`` or ``'bravyi_kitaev'``.
+
+    Returns:
+       pennylane.operator: Hamiltonian for the Fermi-Hubbard model.
+
+    **Example**
+
+    >>> n_cells = [2,2]
+    >>> h = [[0.5]]
+    >>> u = [[1.0]]
+    >>> spin_ham = fermihubbard("Square", n_cells, hopping=h, interaction=u)
+    >>> print(spin_ham)
+
+    """
+
+    lattice = _generate_lattice(lattice, n_cells, boundary_condition, neighbour_order)
+
+    if hopping is None:
+        hopping = [1.0]
+    elif isinstance(hopping, (int, float, complex)):
+        hopping = [hopping]
+
+    hopping = math.asarray(hopping)
+
+    if hopping.shape not in [(neighbour_order,), (lattice.n_sites, lattice.n_sites)]:
+        raise ValueError(
+            f"hopping should be a number or an array of shape {neighbour_order}x1 or {lattice.n_sites}x{lattice.n_sites}"
+        )
+
+    spin = 2
+    hopping_ham = 0.0
+    if hopping.shape == (neighbour_order,):
+        for edge in lattice.edges:
+            for s in range(spin):
+                i, j, order = edge[0], edge[1], edge[2]
+                s1 = i * spin + s
+                s2 = j * spin + s
+                hopping_term = -hopping[order] * (
+                    FermiWord({(0, s1): "+", (1, s2): "-"})
```

Comment:
We should add a note in the docstring regarding the convention of the sites uses - up-down-up-down

Reply:
Good point, will do.